apps/remote_control/lib/lexical/remote_control/code_action/handlers/remove_unused_alias.ex

```diff
@@ -0,0 +1,191 @@
+defmodule Lexical.RemoteControl.CodeAction.Handlers.RemoveUnusedAlias do
+  @moduledoc """
+  A code action that removes an unused alias
+
+  Most of the code actions are fairly straightforward, but I think this one deserves a couple of comments on
+  the approach. I initially tried the following:
+
+    * Finding the alias via Sourceror's Zipper.find
+    * Rewriting the ast via Macro.prewalk / postwalk
+    * Using Macro.travesrse where I'd mark the metadata as being deleted in the prewalker, and
+      delete it in the postwalker.
+
+  They had the following problems.
+  Sourceror would consistently produce ast that was not recognized by elixir 1.14's code normalizer, causing a crash.
+  Using AST rewriting was susceptible to infinite recursion, and it was extremely difficult to delete blocks reliably.
+  Blocks in one context would be deleted, but with a different formulation, nils would appear in the output code.
+  It was also very difficult to pop up the stack and delete an entire multiple alias without zippers.
+
+  So the approach we have here utilizes a hybrid of AST walking / text replacement. It works for all the examples
+  I could come up with, but it's a bit longer than I desired. Dorgan said he'd take a look at the errors in the
+  normalizer and possibly fix sourceror, so until then, this is what we have.
+  """
+
+  alias Lexical.Ast
+  alias Lexical.Ast.Analysis
+  alias Lexical.Document
+  alias Lexical.Document.Changes
+  alias Lexical.Document.Edit
+  alias Lexical.Document.Position
+  alias Lexical.Document.Range
+  alias Lexical.RemoteControl.Analyzer
+  alias Lexical.RemoteControl.CodeAction
+  alias Lexical.RemoteControl.CodeAction.Diagnostic
+
+  import Record
+
+  defrecordp :multi_alias_metadata, [
+    :document,
+    :multi_alias_range,
+    :removed_alias_range,
+    :alias_count
+  ]
+
+  defrecordp :single_alias_metadata, [:document, :range]
+  @behaviour CodeAction.Handler
+
+  @impl CodeAction.Handler
+  def actions(%Document{} = document, %Range{} = range, diagnontics) do
+    Enum.reduce(diagnontics, [], fn %Diagnostic{} = diagnostic, acc ->
+      case to_edit(document, range.start, diagnostic) do
+        {:ok, module_name, edit} ->
+          changes = Changes.new(document, [edit])
+          action = CodeAction.new(document.uri, "Remove alias #{module_name}", :source, changes)
+
+          [action | acc]
+
+        _ ->
+          acc
+      end
+    end)
+  end
+
+  @impl CodeAction.Handler
+  def kinds do
+    [:source]
+  end
+
+  @alias_regex ~r/unused alias (\w+)/
+  defp to_edit(%Document{} = document, %Position{} = position, %Diagnostic{} = diagnostic) do
+    with [[_, module_string]] <- Regex.scan(@alias_regex, diagnostic.message),
```

Comment:
Recommend splitting out a function for this:

```elixir
  @alias_regex ~r/unused alias (\w+)/
  defp fetch_unused_alias(diagnostic) do
    case Regex.scan(@alias_regex, diagnostic.message) do
      [[_, module_string]] -> {:ok, String.to_atom(module_string)}
      _ -> :error
    end
  end
```

```suggestion
    with {:ok, last_segment} <- fetch_unused_alias(diagnostic),
```